File: LeetCode/Algorithm/283. Move Zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        
        count = len(nums)
        index = 0
        while count > 0 :
            count -= 1
            if nums[index] == 0:
                nums.append(nums.pop(index))
            else :
                index += 1

File: LeetCode/Algorithm/283. Move Zeroes.cs
public class Solution {
    public void MoveZeroes(int[] nums) {
        var slow = 0;
        var fast = 0;
        while (fast < nums.Length)
        {
            if (nums[slow] == 0 && nums[fast] != 0)
            {
                var temp = nums[fast];
                nums[fast] = nums[slow];
                nums[slow] = temp;
            }
                
            if (nums[slow] != 0)
                slow += 1;
            fast += 1;
        }
    }
}